pkg/virt-handler/virtwrap/manager.go

```diff
@@ -61,6 +63,48 @@ type LibvirtDomainManager struct {
 	podIsolationDetector isolation.PodIsolationDetector
 }
 
+func (l *LibvirtDomainManager) CreateNetwork(netXML *string, netName string) error {
+	network, err := l.virConn.LookupNetworkByName(netName)
+	if err != nil {
+		network, err = l.virConn.NetworkDefineXML(*netXML)
+		if err != nil {
+			logging.DefaultLogger().Error().Reason(err).Msgf("defining network from xml: %s", *netXML)
+			return err
+		}
+	}
+
+	err = network.SetAutostart(true)
+	if err != nil {
+		logging.DefaultLogger().Error().Reason(err).Msgf("Setting network to autostart: %s", *netXML)
+
+		return goerrors.New(fmt.Sprintf("Cannot set network %s to autostart", netName))
+	}
+
+	active, err := network.IsActive()
+	if err != nil || !active {
+		err = network.Create()
+		if err != nil {
+			return goerrors.New(fmt.Sprintf("Network %s creation failed.", netName))
+		}
+	}
+
+	return nil
+}
+func (l *LibvirtDomainManager) DestroyNetwork(netName string) error {
+	network, err := l.virConn.LookupNetworkByName(netName)
+	if err != nil {
```

Comment:
err doesn't necessarily mean the network wasn't found.

You could do something like this. 

```
if domainerrors.IsNotFound(err) {
```